drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -56,6 +70,26 @@ local function find_default_endpoint(device, component)
   return device.MATTER_DEFAULT_ENDPOINT
 end
 
+local function assign_child_profile(device, child_ep)
+  local profile
+  for _, ep in ipairs(device.endpoints) do
+    if ep.endpoint_id == child_ep then
+      -- Some devices report multiple device types which are a subset of
+      -- a superset device type (For example, Dimmable Light is a superset of
+      -- On/Off light). This mostly applies to the four light types, so we will want
+      -- to match the profile for the superset device type. This can be done by
+      -- matching to the device type with the highest ID
+      local id = 0
+      for _, dt in ipairs(ep.device_types) do
+        id = math.max(id, dt.device_type_id)
+        profile = device_type_profile_map[id]
```

Comment:
quibble, but you could do this outside the loop, right?

Reply:
Good point, moved the assignment outside the loop 👍